### PR TITLE
Add battery-full icon

### PR DESCRIPTION
**Pull request by M-arcus**

### CODE REVIEW

**Comment by herohamp** [comment]
This would be a great inclusion

**Comment by locness3** [comment]
#171 Guidelines says "no fills"

File: icons/battery-full.svg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="#000" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
+    <rect x="1" y="6" width="18" height="12" rx="2" ry="2"></rect>
+    <line x1="23" y1="13" x2="23" y2="11"></line><rect x="5" y="10" width="10" height="4" style="fill:#000"></rect>
```
Comment (locness3) [inline icons/battery-full.svg]: `<line>`tag is colsing with `</rect>` :thinking:

**Comment by Ecleptic** [comment]
Looks like I completely missed that in the guidelines. I'll re-build it. 
I need a battery in my current project.

**Comment by ashygee** [comment]
@herohamp curious why this is approved? the icon still contains a fill which is against the guidelines (#171) pointed out by @locness3.

**Comment by ashygee** [comment]
cc @colebemis

**Comment by M-arcus** [comment]
@locness3 Sorry, I can't see the fill. Only that the fill is set to `none`.

Could you please point it out?

**Comment by locness3** [comment]
@M-arcus Strange, the Github rich diff displays a fill
screenshot_2019-02-15 add battery-full icon by m-arcus pull request 172 feathericons feather

**Comment by M-arcus** [comment]
@locness3 Probably because the line are thick. Should i make them thinner?

**Comment by locness3** [comment]
> @locness3 Probably because the line are thick. Should i make them thinner?

Guidelines say (#171) that icons should have a 2px stroke. Check out the guidelines about spacing.

**Comment by mittalyashu** [comment]
The battery looks great, can you add 1px of spacing around the icon.

Can you create a different version of battery like:

| Battery 0 | Battery 1 | Battery 2 | Battery 3 |
| --- | --- | --- | --- |
| Figma_9Lqtq9H4sE | Figma_B31Unm4d5q | Figma_uaKYg3v7DM | Figma_mrUd1wBnYs |

**Comment by csandman** [comment]
I know this is old, but I wanted to weigh in.  I feel like the version by @mittalyashu doesn't quite match the guidelines for 2px wide stroke width's with the bars.  I feel like the best solution to this would be to implement a similar shape but using lines instead.  Here are my versions that stick with the original battery's style (using rect and line):

---
`battery-full`

image

```xml
<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <rect x="1" y="6" width="18" height="12" rx="2" ry="2" />
  <line x1="23" y1="13" x2="23" y2="11" />
  <line x1="5" y1="14" x2="5" y2="10" />
  <line x1="10" y1="14" x2="10" y2="10" />
  <line x1="15" y1="14" x2="15" y2="10" />
</svg>
```
---
`battery-medium`

image

```xml
<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <rect x="1" y="6" width="18" height="12" rx="2" ry="2" />
  <line x1="23" y1="13" x2="23" y2="11" />
  <line x1="5" y1="14" x2="5" y2="10" />
  <line x1="10" y1="14" x2="10" y2="10" />
</svg>
```
---
`battery-low`

image

```xml
<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <rect x="1" y="6" width="18" height="12" rx="2" ry="2" />
  <line x1="23" y1="13" x2="23" y2="11" />
  <line x1="5" y1="14" x2="5" y2="10" />
</svg>
```

And here are the files: 
[battery-icons.zip](https://github.com/feathericons/feather/files/4548333/battery-icons.zip)

I could submit a separate pull request, what do you think @colebemis?  I know you've been busy so I tried to make these match the current style (code and guidelines) as much as possible to make it easy for you haha.

**Comment by M-arcus** [comment]
Completed via https://github.com/lucide-icons/lucide/pull/23

**Pull request closed without merging**